guides/keras_nlp/fnet_classification.py

```diff
@@ -0,0 +1,380 @@
+introduced the Transformer architecture. The Transformer model uses self-attention
+to learn representations of tokens in a piece of text. Succinctly put, to
+compute the representation of a token, the self-attention mechanism computes
+attention scores between a token and every other token in the sequence.
+It then uses the scores to compute a weighted average of the tokens in the sequence.  
+Since then, multiple language models such as BERT, RoBERTa, etc. have
+been released. All these models have the same Transformer architecture with 
+different pretraining tasks in order to "learn" language.
+
+Note: To learn more about the Transformer architecture, please visit [Jay Alammar's
+peerless blog](https://jalammar.github.io/illustrated-transformer/). 
+
+Recently, there has been an effort to reduce the time complexity of the self-attention
+mechanism and improve performance without sacrificing the quality of results.
+Models such as Longformers, Reformers, etc. come to mind. In 2020, a paper titled
+[FNet: Mixing Tokens with Fourier Transforms](https://arxiv.org/abs/2105.03824)
+replaced the self-attention layer in BERT with a Fourier Transform layer. The
+magnitude of this was not lost on the research community; replacing a relatively
+complicated self-attention layer with a simpler "token mixing" layer resulted in
+comparable accuracy and a speed-up during training. This opens up further avenues
+of research - can we replace the self-attention layer with any simple "token mixing"
+layer and get comparable results?
+
+A couple of points from the paper stood out:
+1. The authors claim that FNet is 80% faster than BERT on GPUs and 70% faster on TPUs.
+The reason for this speed-up is two-fold:
+    a. The Fourier Transform layer is unparametrized, it does not have any parameters!
+    b. The authors use Fast Fourier Transform (FFT); this reduces the time complexity
+    from O(n^2) (in the case of self-attention) to O(n log n).
+2. What's astounding is that FNet still manages to achieve 92-97% of the accuracy
+of BERT on the GLUE benchmark.
+"""
+
+"""
+## Setup
+
+Before we start with the implementation, let's install the KerasNLP library, and
+import all the necessary packages.
+"""
+
+"""shell
+pip install -q keras-nlp
+pip install -q tfds-nightly
+"""
+import keras_nlp
+import numpy as np
+import pathlib
+import random
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+from tensorflow import keras
+from tensorflow_text.tools.wordpiece_vocab import bert_vocab_from_dataset as bert_vocab
+
+"""
+Let's also define our parameters/hyperparameters.
+"""
+BATCH_SIZE = 64
+EPOCHS = 3
+MAX_SEQUENCE_LENGTH = 40
+VOCAB_SIZE = 15000
+
+EMBED_DIM = 128
+INTERMEDIATE_DIM = 512
+
+"""
+## Loading the Dataset
+Tensorflow Datasets (TFDS) is a library that provides a unified API for working
+with datasets that are stored in the TensorFlow format. We will use TFDS to load
+the SST-2 dataset.
+"""
+train_ds, val_ds, test_ds = tfds.load(
```

Comment:
One more thing - let's be more concise in the description, if we choose to use tfds, just say load SST-2 from Tensorflow Datasets (TFDS).

Reply:
This was a request from @fchollet when I was doing my guide, so maybe we should discuss with him? Personally not particularly opinionated.

Reply:
Ah okay, let's bring this up in the team chat. I feel since tfds is part of TF ecosystem and still being maintained, we should try using their product.

Reply:
Removed TFDS. We don't need it for IMDb :)